agent-core/src/main/java/com/newrelic/agent/android/agentdata/AgentDataReporter.java

```diff
@@ -121,6 +122,19 @@ protected void reportCachedAgentData() {
 
     public Future reportAgentData(Payload payload) {
         PayloadSender payloadSender = new AgentDataSender(payload, getAgentConfiguration());
+
+        if (payload.getBytes().length > Constants.Network.MAXPAYLOADSIZELIMIT) {
+            DeviceInformation deviceInformation = Agent.getDeviceInformation();
+            String name = MetricNames.SUPPORTABILITY_MAXPAYLOADSIZELIMIT_ENDPOINT
+                    .replace(MetricNames.TAG_FRAMEWORK, deviceInformation.getApplicationFramework().name())
+                    .replace(MetricNames.TAG_DESTINATION, MetricNames.METRIC_DATA_USAGE_COLLECTOR)
+                    .replace(MetricNames.TAG_SUBDESTINATION, "f");
+            StatsEngine.get().inc(name);
```

Comment:
When recording supportability metrics, use the [SUPPORTABILITY](https://github.com/newrelic/newrelic-android-agent/blob/develop/agent-core/src/main/java/com/newrelic/agent/android/stats/StatsEngine.java#L24) instance or [notice()](https://github.com/newrelic/newrelic-android-agent/blob/develop/agent-core/src/main/java/com/newrelic/agent/android/stats/StatsEngine.java#L38) method. This is an subclass that decorates the metric with some of the fields you referenced here.

Reply:
Fixed